### DISCUSSION TITLE
Wordpress Helm Release failed

### DISCUSSION THREAD

**Discussion by Mel-Github** [fluxcd/flux2#1205] [Q&A] [answered]
Was trying out flux with bitnami wordpress Helm Charts. The HelmRelease is always unable to complete the reconcile process. However, the application is successfully deployed behind the scene. Anyone?

`flux create helmrelease wordpress --source=HelmRepository/bitnami --chart wordpress --release-name wordpress --target-namespace default --interval 5m0s`
```
✚ generating HelmRelease
► applying HelmRelease
✔ HelmRelease updated
◎ waiting for HelmRelease reconciliation
✗ context deadline exceeded
```

`flux get helmrelease`
```
NAME     	READY	MESSAGE                         	REVISION	SUSPENDED
redis    	True 	Release reconciliation succeeded	12.10.0 	False
wordpress	False	upgrade retries exhausted       	        	False
```

`k get all -l app.kubernetes.io/instance=wordpress`

```
NAME                             READY   STATUS    RESTARTS   AGE
pod/wordpress-59f4d75675-7bhq8   1/1     Running   0          15m
pod/wordpress-mariadb-0          1/1     Running   0          15m

NAME                        TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
service/wordpress           LoadBalancer   10.105.80.146   <pending>     80:30706/TCP,443:31570/TCP   15m
service/wordpress-mariadb   ClusterIP      10.99.53.101    <none>        3306/TCP                     15m

NAME                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/wordpress   1/1     1            1           15m

NAME                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/wordpress-59f4d75675   1         1         1       15m

NAME                                 READY   AGE
statefulset.apps/wordpress-mariadb   1/1     15m
```

**Comment by stefanprodan** [comment, marked as answer]
Your cluster doesn’t have a load balancer so Helm times out waiting for the an external IP. You should switch to NodePort or disable resource waiting https://toolkit.fluxcd.io/components/helm/helmreleases/#disabling-resource-waiting 

Note that disabling waiting means that no health checks are performed, if the deployment fails you’ll not receive notifications, nor will you be able to use helm-controller automated rollback.

**Comment by Mel-Github** [comment]
Thanks Stefan...that make senses..
